giskard/ml_worker/utils/cache.py

```diff
@@ -0,0 +1,28 @@
+import hashlib
+
+
+class SimpleCache:
+    def __init__(self, max_results=20):
+        self.results = {}
+        self.order = []  # To maintain the order in which results were added
+        self.max_results = max_results
+
+    def add_result(self, obj, result):
+        # Calculate the hash of the object
+        obj_hash = hashlib.md5(repr(obj).encode()).hexdigest()
+
+        # Store the result with the object's hash as the key
+        self.results[obj_hash] = result
+
+        # Add the object's hash to the order list and remove the oldest result if necessary
+        self.order.append(obj_hash)
```

Comment:
I think you should use LRU (least recently used) algorithm here, instead of oldest.